internal/probe/webserver.go

```diff
@@ -53,20 +58,32 @@ func performWebServerProbe(targets []string) ([]URLDetails, []string, error) {
 	}
 	defer httpxRunner.Close()
 
-	httpxRunner.RunEnumeration()
+	// Run the enumeration with a goroutine and select for timeout
+	done := make(chan struct{})
 
-	return urls, errors, nil
+	go func() {
+		httpxRunner.RunEnumeration()
+		close(done)
+	}()
 
+	select {
+	case <-ctx.Done():
```

Comment:
will ctx.Done() trigger if the enumeration was completed successfully? should we move it to after the done case?

Reply:
Since its a case statement it will only get triggered on a failure

> Done returns a channel that's closed when work done on behalf of this context should be canceled. Done may return nil if this context can never be canceled. Successive calls to Done return the same value. The close of the Done channel may happen asynchronously, after the cancel function returns.